src/instructlab/model/backends/backends.py

```diff
@@ -113,19 +113,22 @@ def is_model_gguf(model_path: pathlib.Path) -> bool:
         return first_four_bytes_int == GGUF_MAGIC
 
 
-def determine_backend(model_path: pathlib.Path) -> str:
+def determine_backend(model_path: pathlib.Path) -> Tuple[str, str]:
     """
     Determine the backend to use based on the model file properties.
     Args:
         model_path (Path): The path to the model file.
     Returns:
-        str: The backend to use.
+        Tuple[str, str]: A tuple containing two strings:
+                        - The backend to use.
+                        - The reason why the backend was selected.
     """
     # Check if the model is a GGUF file
 
     # If the model is a directory, it's a VLLM model - it's kinda weak, but it's a start
     if model_path.is_dir() and sys.platform == "linux":
-        return VLLM
+        logger.debug(f"Model is a directory and system is Linux, using {VLLM} backend.")
+        return VLLM, "model is a directory and running on Linux."
```

Comment:
perhaps we should be checking for the existence of .safetensors files instead of just checking if the supplied model is a directory
this could be fixed in a follow up PR though

Reply:
nvm, @alimaredia pointed me to https://github.com/instructlab/instructlab/issues/1457

Reply:
Today, we don't check for the tensors files, just for a directory, there is an improvement filed here https://github.com/instructlab/instructlab/issues/1457.

However I'll make it clearer.